canonical/maas-ui
# feat(machines): select machines using the new types

## Done

- Update the checkboxes in the machine list to select filters, groups and machines.

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Go to the machine list.
- Tick some machines.
- Open the redux dev tools and look at `machine` -> `selectedMachines` and you should see the machine ids.
- Tick a group and you should see the group in the selected state (not all grouping is currently working so you might have to change the table grouping to find one that the API supports - owner was working for me).
- Tick the checkbox in the table header (next to FQDN) and the selected state should show the filter (probably empty unless you've set one).

## Fixes

Fixes: canonical/app-tribe#1126.
Fixes: canonical/app-tribe#1257.
